### PR TITLE
Fix go version and move send to SendAsync

### PR DESCRIPTION
**Pull request by tuteng**
* Update native-go-client to latest
* Move sync send to async send
* Add config DisableBatching and BatchingMaxPublishDelay

### CODE REVIEW

File: pulsar/config.go [hunk 1]
```diff
@@ -53,8 +53,8 @@ type pulsarConfig struct {
 	BlockIfQueueFull                   bool                      `config:"block_if_queue_full"`
 	HashingScheme                      pulsar.HashingScheme      `config:"hashing_schema"`
 	CompressionType                    pulsar.CompressionType    `config:"compression_type"`
-	Batching                           bool                      `config:"batching"`
```
Comment (sijie) [inline pulsar/config.go:56]: this is a breaking change. Can we keep this and mark it as deprecated?

File: pulsar/config.go [hunk 2]
```diff
@@ -53,8 +53,8 @@ type pulsarConfig struct {
...
-	BatchingMaxPublishDelay            time.Duration             `config:"batching_max_publish_delay"`
+	DisableBatching                           bool                      `config:"disable_batching"`
+	BatchingMaxPublishDelay            int             `config:"batching_max_publish_delay"`
```
Comment (sijie) [inline pulsar/config.go]: This is a breaking change. Do we really need to change the type?

File: vendor/github.com/apache/pulsar-client-go/examples/consumer-listener/consumer-listener.go [hunk 3]
```diff
@@ -18,47 +18,47 @@
 package main
 
 import (
-    "fmt"
-    "log"
+	"fmt"
```
Comment (sijie) [inline vendor/github.com/apache/pulsar-client-go/examples/consumer-listener/consumer-listener.go:21]: It seems that there are many format changes. Can we do the format changes in a separate pull request?

**Review by sijie — CHANGES_REQUESTED** [review]

**Comment by sijie** [comment]
need a pulsar-client-go release.

**Comment by tuteng** [comment]
Close this pr, it has been fixed #20

**Pull request closed without merging**